### DISCUSSION TITLE
Work with (Cloudinary) media AND without a server

### DISCUSSION THREAD

**Discussion by ianwensink** [tinacms/tinacms#2338] [Help] [answered]
It seems as if Tina doesn't require a running server/serverless, and can be fully used with a static build; as long as you're using Tina Cloud. That's awesome!

When I want to setup media, though, the guide tells me to create a serverless api for Cloudinary. I reckon this is for authentication and such.

My question is: is this necessary? It would be awesome for my use case to host a Tina-powered app without a server. I guess the authorization with Cloudinary would be through Tina Cloud as well, just as it is with content?

Is this possible? Or is this on the roadmap somehow?

Also: Cloudinary is not a must for me, Github files would work too, but it seems as per [this discussion](https://github.com/tinacms/tinacms/discussions/2298) that it is not supported to host media through the Github repo?

**Comment by logan-anderson** [comment, marked as answer]
@ianwensink This is a great question! 

> My question is: is this necessary? It would be awesome for my use case to host a Tina-powered app without a server. I guess the authorization with Cloudinary would be through Tina Cloud as well, just as it is with content?

yes this is _currently_ necessary we do not have any current plans in our roadmap to support proxying Cloudinary though Tina Cloud (you don't need your own server) but this is on our minds. In an ideal world no backend is needed to edit your content.

> Also: Cloudinary is not a must for me, Github files would work too, but it seems as per this discussion that it is not supported to host media through the Github repo?

We have had a lot of requests for "Github backed media" this is on our minds and will be the next media source we tackle for Tina Cloud.  This is not on our [current roadmap](https://github.com/tinacms/tinacms/discussions/2255) but it is definilty something we want to work on in 2022.

**Reply by ianwensink** [reply]
@logan-anderson Thank you for your reply!

So in conclusion it is currently **required** to host a Node server, when working with Tina combined with the media store.

I'm looking forward to a solution without a server!

**Reply by logan-anderson** [reply]
@ianwensink yeah exactly. we currently recommend  a next.js API function. 
See the docs here: https://tina.io/docs/media-cloudinary/
see an example here: https://github.com/tinacms/tina-cloud-starter/blob/main/pages/api/cloudinary/%5B...media%5D.tsx

**Reply by xxgeoffreyxx** [reply]
What about using other CDN's such as S3? Is Cloudinary "baked in" as it were?

**Reply by cloudcompute** [reply]
No further discussion on this topic. Looks like either Cloudinary is promoting Tina Or the other way round.

If one is already paying to S3 then why should one pay to Cloudinary.  As far as I am concerned, I'd surely go with other CMS.

**Reply by perkinsjr** [reply]
We have an open-source DAM that we are exploring/working on before we launch 1.0 that would allow a user to not have to use Cloudinary at all. This has been a discussion point since late 2021. 

> No further discussion on this topic. Looks like either Cloudinary is promoting Tina Or the other way round.

We decided on Cloudinary for its free tier and it gave us the ability to move quickly at the beginning working on more of the core product.

**Reply by cloudcompute** [reply]
Things are clear now.
